MapleServer2/Types/Club.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using MaplePacketLib2.Tools;
+using MapleServer2.Packets;
+using MapleServer2.Servers.Game;
+using MapleServer2.Tools;
+
+namespace MapleServer2.Types
+{
+    public class Club
+    {
+        public long Id { get; }
+        public string Name { get; set; }
+        public bool Approval { get; set; } //Require approval before someone can join
+        public Player Leader { get; set; }
+        public int MaxMembers { get; set; }
+
+        //List of players and their session.
+        public List<Player> Members { get; }
+
+        public Club(List<Player> cPlayers)
+        {
+            Id = GuidGenerator.Long();
+            MaxMembers = 10;
+            Leader = cPlayers.First();
+            Members = cPlayers;
+            Approval = true;
+        }
+        public Club(Party party, string name)
+        {
+            Id = GuidGenerator.Int();
+            Name = name;
+            MaxMembers = 10;
+            Leader = party.Leader;
+            Members = party.Members;
```

Comment:
I know I said to do this but on second thought, setting the members of the club immediately to the members of the party probably doesn't make sense. You can keep this line of code though for now, just a heads up though that we probably want only the leader to be added as the only member of the club at first